### PR TITLE
Implement EnableWideBitFieldTypes. Fixes #74.

### LINKED ISSUE #74: bit field has invalid type (have unsigned long long)
```
/usr/include/linux/perf_event.h:309:23: bit field has invalid type (have unsigned long long)
```

<details>

```c
struct perf_event_attr {
	unsigned long long			disabled       :  1, /* off by default        */
				inherit	       :  1, /* children inherit it   */
				pinned	       :  1, /* must always be on PMU */
				exclusive      :  1, /* only group on PMU     */
				exclude_user   :  1, /* don't count user      */
				exclude_kernel :  1, /* ditto kernel          */
				exclude_hv     :  1, /* ditto hypervisor      */
				exclude_idle   :  1, /* don't count when idle */
				mmap           :  1, /* include mmap data     */
				comm	       :  1, /* include comm data     */
				freq           :  1, /* use freq, not period  */
				inherit_stat   :  1, /* per task counts       */
				enable_on_exec :  1, /* next exec enables     */
				task           :  1, /* trace fork/exit       */
				watermark      :  1, /* wakeup_watermark      */
				/*
				 * precise_ip:
				 *
				 *  0 - SAMPLE_IP can have arbitrary skid
				 *  1 - SAMPLE_IP must have constant skid
				 *  2 - SAMPLE_IP requested to have 0 skid
				 *  3 - SAMPLE_IP must have 0 skid
				 *
				 *  See also PERF_RECORD_MISC_EXACT_IP
				 */
				precise_ip     :  2, /* skid constraint       */
				mmap_data      :  1, /* non-exec mmap data    */
				sample_id_all  :  1, /* sample_type all events */

				exclude_host   :  1, /* don't count in host   */
				exclude_guest  :  1, /* don't count in guest  */

				exclude_callchain_kernel : 1, /* exclude kernel callchains */
				exclude_callchain_user   : 1, /* exclude user callchains */
				mmap2          :  1, /* include mmap with inode data     */
				comm_exec      :  1, /* flag comm events that are due to an exec */
				use_clockid    :  1, /* use @clockid for time fields */
				context_switch :  1, /* context switch data */
				write_backward :  1, /* Write ring buffer from end to beginning */
				__reserved_1   : 36;
}
```
</details>

Fix incoming.

### PR DESCRIPTION
**Pull request by pwaller**
Fixes #74.

Conflicts with #73 which should be merged first, then I will rebase this one.

But this can be reviewed now in case modifications are required.

### CODE REVIEW

File: ast2.go [hunk 1]
```diff
@@ -2136,6 +2140,13 @@ func (n *StructDeclarator) post(lx *lexer) {
 			switch t.Kind() {
 			case Char, SChar, UChar, Int, UInt, Long, ULong, Short, UShort, Enum, Bool:
 				// ok
+			case LongLong, ULongLong:
+				if lx.tweaks.enableLongLongBitFieldTypes {
+					// Non-standard, but enabled.
+					break
+				}
+				lx.report.Err(n.ConstantExpression.Pos(), "bit field has invalid type (have %s)", t)
```
Comment (cznic) [inline ast2.go]: You can use `fallthrough` here, if you like.

File: testdata/issue74.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
+struct perf_event_attr {
```
Comment (cznic) [inline testdata/issue74.h]: This code has copyright, but it's not reproduced here. Even then, we would need to include the respective LICENSE. Perfectly possible, of course.

Or you can just write your own `issue74.h` from scratch, if you like.

Comment (pwaller) [inline testdata/issue74.h]: Will make one.

File: ast2.go (hunk 1 above)
Comment (cznic) [inline ast2.go]: It's okay as it is now.

File: testdata/issue74.h (hunk 2 above)
Comment (cznic) [inline testdata/issue74.h]: OK, thanks.

**Comment by pwaller** [comment]
Ready once #73 is merged and I've rebased.

**Comment by pwaller** [comment]
Ah wait, I'll rename `EnableLongLongBitFieldTypes` to `EnableWideBitFieldTypes` for consistency with `EnableWideEnumValues`.

**Comment by pwaller** [comment]
Rebased, tests pass. Merging.

**Comment by cznic** [comment]
:+1:

**Pull request merged**